lib/Dialect/FIRRTL/FIRRTLFolds.cpp

```diff
@@ -1638,16 +1638,38 @@ struct FoldNodeName : public mlir::RewritePattern {
         !node.annotations().empty())
       return failure();
     auto *expr = node.input().getDefiningOp();
-    if (expr && !expr->hasAttr("name") && !isUselessName(name))
+    // Best effort
+    if (name && expr && !expr->hasAttr("name"))
       rewriter.updateRootInPlace(expr, [&] { expr->setAttr("name", name); });
+    // Make sure we don't revisit this node.  This is redundant with NodeBypass,
+    // but easier to do here.
     rewriter.replaceOp(node, node.input());
+    rewriter.eraseOp(node);
```

Comment:
I think this is erasing `node` twice.

Reply:
Fixed by https://github.com/llvm/circt/commit/f69779d2cb5952c4fc402fcd32e6dbcc9e9582f6

Reply:
I was trying to track this last night.  I wasn't seeing replaceOp erasing the old node.

Reply:
I think `op.replaceOp` doesn't erase `op` but `rewriter.replaceOp` erases `op`.

https://github.com/llvm/llvm-project/blob/9f2b873a7de78c191811d03fd6b30cfc5c04505e/mlir/lib/IR/PatternMatch.cpp#L245-L263